frontend/public/components/utils/get-pod-traffic.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { Tooltip } from '@patternfly/react-core';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { EndpointSliceKind } from '@console/internal/module/k8s';
+import { ConnectedIcon, DisconnectedIcon } from '@patternfly/react-icons';
+import { EndPointSliceModel } from '@console/app/src/models';
+
+export type GetPodTrafficProp = {
+  podName: string;
+  namespace: string;
+  tooltipFlag?: boolean;
+};
+
+export const GetPodTraffic: React.FC<GetPodTrafficProp> = ({ podName, namespace, tooltipFlag }) => {
+  const { t } = useTranslation();
+  const [data, loaded, loadError] = useK8sWatchResource<EndpointSliceKind[]>({
+    groupVersionKind: {
+      kind: EndPointSliceModel.kind,
+      version: EndPointSliceModel.apiVersion,
+    },
+    isList: true,
+    namespaced: true,
+    namespace,
+  });
+
+  const allEndpoints = data.reduce((prev, next) => prev.concat(next?.endpoints), []);
+  const receivingTraffic = allEndpoints.find((obj) => obj?.targetRef?.name === podName);
+
+  if (tooltipFlag) {
```

Comment:
do we need this i.e when we need tooltips and when we don't? even in the list, we can show the tooltip right?

Reply:
Tooltip is required only for topology side bar in pod details section. So not added tooltip for list pages. This is based on the conversation - https://coreos.slack.com/archives/CHGU4P8UU/p1668540290287939?thread_ts=1668424861.378819&cid=CHGU4P8UU